### PR TITLE
Bug 1882630: fix stale lsblk output

### PR DESCRIPTION
**Pull request by sp98**
lsblk, sometimes, reports stale fstype information. Using blkid to fetch fstype information for the devices

Signed-off-by: Santosh Pillai <sapillai@redhat.com>

### CODE REVIEW

**Comment by gnufied** [comment]
@sp98 can you link BZ to this PR?

File: pkg/diskmaker/discovery/discovery_test.go [hunk 1]
```diff
@@ -13,27 +13,48 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+var lsblkOut string
+var blkidOut string
+
+// HelperCommand returns a fake exec.Cmd for unit tests
+func HelperCommand(command string, args ...string) *exec.Cmd {
```
Comment (gnufied) [inline pkg/diskmaker/discovery/discovery_test.go:20]: does this has to be an exported function?

File: pkg/internal/diskutil_test.go [hunk 2]
```diff
@@ -11,35 +11,59 @@ import (
...
 `
-	lsblkOutput2 = `NAME="sdc" KNAME="sdc" ROTA="1" TYPE="disk" SIZE="62914560000" MODEL="VBOX HARDDISK" VENDOR="ATA" RO="0" RM="1" STATE="running" FSTYPE="ext4" SERIAL=""
-NAME="sdc3" KNAME="sdc3" ROTA="1" TYPE="part" SIZE="62913494528" MODEL="" VENDOR="" RO="0" RM="1" STATE="" FSTYPE="ext4" SERIAL=""
+	blkIDOutput1 = `/dev/sdc: TYPE="ext4"
+/dev/sdc3: TYPE="ext2"
 `
 )
 
+// HelperCommand returns a fake exec.Cmd for unit tests
+func HelperCommand(command string, args ...string) *exec.Cmd {
```
Comment (gnufied) [inline pkg/internal/diskutil_test.go:30]: same as above

File: pkg/internal/diskutil_test.go [hunk 3]
```diff
@@ -11,35 +11,59 @@ import (
...
-	fmt.Fprintf(os.Stdout, os.Getenv("STDOUT"))
+	switch os.Getenv("COMMAND") {
+	case "lsblk":
+		fmt.Fprintf(os.Stdout, os.Getenv("LSBLKOUT"))
+	case "blkid":
+		fmt.Fprintf(os.Stdout, os.Getenv("BLKIDOUT"))
+	}
 }
 
 func TestListBlockDevices(t *testing.T) {
```
Comment (gnufied) [inline pkg/internal/diskutil_test.go:54]: unnecessary new line

File: pkg/internal/diskutil_test.go [hunk 4]
```diff
@@ -113,16 +138,17 @@ func TestListBlockDevices(t *testing.T) {
 		},
 		{
 			label:             "Case 3",
```
Comment (gnufied) [inline pkg/internal/diskutil_test.go:140]: please change labels to be descriptive everywhere. "Case 1".. etc are not good enough labels.

**Review by gnufied — COMMENTED** [review]
Mostly looks good to me. some minor and style related comments.

File: pkg/diskmaker/discovery/discovery_test.go (hunk 1 above)
Comment (sp98) [reply pkg/diskmaker/discovery/discovery_test.go]: it doesn't. Updated it to be non-exported function.

File: pkg/internal/diskutil_test.go (hunk 2 above)
Comment (sp98) [reply pkg/internal/diskutil_test.go]: Updated it to be non-exported function.

File: pkg/internal/diskutil_test.go (hunk 3 above)
Comment (sp98) [reply pkg/internal/diskutil_test.go]: removed.

File: pkg/internal/diskutil_test.go (hunk 4 above)
Comment (sp98) [reply pkg/internal/diskutil_test.go]: added descriptive labels.

**Pull request merged**